### PR TITLE
Generate settings.xml if MAVEN_MIRROR_URL is set

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### What does this PR do?
Generates  `/home/theia/.m2/settings.xml` if `MAVEN_MIRROR_URL` is set.

```xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <mirrors>
    <mirror>
      <url>${env.MAVEN_MIRROR_URL}</url>
      <mirrorOf>external:*</mirrorOf>
    </mirror>
  </mirrors>
</settings>
```

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14391

### CODE REVIEW

**Comment by tolusha** [comment]
@benoitf 
Could you validate the PR ?

**Comment by benoitf** [comment]
@tolusha why do we have M2 settings in theia endpoint ? it's not related to any programming language

**Comment by benoitf** [comment]
Probably you're requiring like `/entrypoint.sh` including optional like `/before-start.sh` and include that file in the java endpoint containers ?

**Comment by tolusha** [comment]
@benoitf 
I've updated the PR

File: dockerfiles/remote-plugin-java11/etc/before-start.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
```
Comment (benoitf) [inline dockerfiles/remote-plugin-java11/etc/before-start.sh]: minor: just 2019

**Pull request merged**